.github/CODE_STANDARDS.md

```diff
@@ -0,0 +1,324 @@
+
+This checklist serves as a guide to both authors and reviewers to ensure a thorough and effective code review process.
+
+## 3. Contributing
+
+### 3.1 Pull Requests && Issues
+
+We have separate templates for Pull Requests and Issues. Please use them when creating a new PR or Issue.
+
+### 3.2 Conventional Commits & Pull Requests Naming
+
+#### 3.2.1 Overview
+
+In an effort to maintain clarity and coherence in our commit history, we are adopting the Conventional Commits style for all commit messages across our repositories. This uniform format not only enhances the readability of our commit history but also facilitates automated tools in generating changelogs and extracting valuable information effectively.
+
+#### 3.2.2 Structure
+
+Conventional Commits follow a structured format: `type(scope): description`, where:
+
+- `type`: Represents the nature of the commit (e.g., feat, fix, chore).
+- `scope`: Denotes the relevant module or issue.
+- `description`: Provides a brief explanation of the change.
+
+When introducing breaking changes, an `!` should be appended after the `type/scope`: `feat(#123)!: introduce a breaking change`.
+
+#### 3.2.3 Types
+
+- `feat`: Utilized when introducing a new feature to the codebase.
+- `fix`: Employed when resolving a bug or issue in the code.
+- `docs`: Designated for commits involving documentation changes, such as updating README files or adding comments.
+- `style`: Applied to commits focusing on code style and formatting, without altering the code's functionality.
+- `refactor`: Used for code changes that neither introduce new features nor fix bugs, but improve the code structure or design.
+- `test`: Assigned to commits pertaining to the addition, modification, or refactoring of tests.
+- `chore`: For changes related to build processes, local development, or other maintenance tasks.
+- `perf`: Employed when enhancing the performance of the codebase.
+- `revert`: Marked for commits that revert a previous change.
+- `ci`: Applied to changes concerning the Continuous Integration (CI) configuration or scripts.
+- `deps`: Used when updating or modifying dependencies.
+
+#### 3.2.4 Conventional Commits - Automatic Versioning
+
+In our repositories, we use Conventional Commits to automatically generate the version number for our releases.
+
+It works like this:
+
+`fix: which represents bug fixes, and correlates to a SemVer patch.`
+`feat: which represents a new feature, and correlates to a SemVer minor.`
+`feat!:, or fix!:, refactor!:, etc., which represent a breaking change (indicated by the !) and will result in a SemVer major.`
+
+Given a version number MAJOR.MINOR.PATCH, increment the:
+
+MAJOR version when you make incompatible API changes
+MINOR version when you add functionality in a backward compatible manner
+PATCH version when you make backward compatible bug fixes
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
```

Comment:
Very well explained, although I really liked my examples with `commit`: `version` -> `version` :)

If you decide to not include them in addition to your explanation, I completely understand, but I will be sad 😢

Reply:
added Kuba - thought I added it earlier, sorry :(